packages/lesswrong/server/karmaChanges.ts

```diff
@@ -1,16 +1,74 @@
 import { Tags } from '../lib/collections/tags/collection';
-import type { KarmaChangeSettingsType } from '../lib/collections/users/schema';
+import {
+  KarmaChangeSettingsType,
+  KarmaChangeUpdateFrequency,
+  karmaChangeNotifierDefaultSettings,
+} from '../lib/collections/users/schema';
 import moment from '../lib/moment-timezone';
 import { compile as compileHtmlToText } from 'html-to-text'
 import sumBy from 'lodash/sumBy';
 import VotesRepo from './repos/VotesRepo';
-import { KarmaChangesArgs } from '../lib/collections/users/karmaChangesGraphQL';
+import type {
+  KarmaChanges,
+  KarmaChangesArgs,
+  PostKarmaChange,
+  CommentKarmaChange,
+  TagRevisionKarmaChange,
+  AnyKarmaChange,
+} from '../lib/collections/users/karmaChangesGraphQL';
+import { isEAForum } from '../lib/instanceSettings';
 
 // Use html-to-text's compile() wrapper (baking in the default options) to make it faster when called repeatedly
 const htmlToTextDefault = compileHtmlToText();
 
 const COMMENT_DESCRIPTION_LENGTH = 500;
 
+const isPostKarmaChange = (change: AnyKarmaChange): change is PostKarmaChange =>
+  "title" in change && !!change.title;
+
+const isCommentKarmaChange = (change: AnyKarmaChange): change is CommentKarmaChange =>
+  "tagCommentType" in change && !!change.tagCommentType;
+
+const getEAKarmaChanges = async (
```

Comment:
I ended up adding a totally different resolver codepath for karma changes on the EA forum. It's much more efficient than the existing resolver and it returns data to the frontend in a format closer to what we actually render.